GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py

```diff
@@ -2,5 +2,5 @@
 
 generator = cms.EDProducer("GenParticles2HepMCConverter",
     genParticles = cms.InputTag("genParticles"),
-    genEventInfo = cms.InputTag("generator"),
+    genEventInfo = cms.InputTag("generator", "", "SIM"),
```

Comment:
the trouble with this solution is that you assume a particular process name that can change (and cause problems if ever this is run together with the generator.. Please change generator to something else. The usual convention is to follow your plugin name

> On Jul 31, 2015, at 2:56 PM, Junghwan John Goh <notifications@github.com> wrote:
> 
> In GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py:
> 
> > @@ -2,5 +2,5 @@
> >  
> >  generator = cms.EDProducer("GenParticles2HepMCConverter",
> >      genParticles = cms.InputTag("genParticles"),
> > -    genEventInfo = cms.InputTag("generator"),
> > +    genEventInfo = cms.InputTag("generator", "", "SIM"),
> 
> Without specifying the process name, it raises some warning because this module tries to read object with same name:
> %MSG-e UnrunnableSchedule: AfterModConstruction 31-Jul-2015 14:51:13 CEST pre-events
> Module run order problem found: 
> generator consumes generator
> Running in the threaded framework would lead to indeterminate results.
> Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
> %MSG
> Of course one can change the name of GenParticles2HepMCConverter module to avoid such overlap, but in this case the input label of RivetAnalyzer have to be changed. It's just a different way to workaround the problem.
> 
> —
> Reply to this email directly or view it on GitHub.
>